### DISCUSSION TITLE
How to traverse  node using variable

### DISCUSSION THREAD

**Discussion by MailNSrinivasan** [schibsted/jslt#257] [Q&A] [answered]
I have a dynamic value in a variable lets say productId. I need to use this variable to traverse a node.

.product.name ==> gives the value of name 

.product.$productId ==> throws error 

but .product."12345" ==> works perfectly. 

Let me know is this possible or am I doing something wrong..

**Comment by larsga** [comment, marked as answer]
So I guess the issue here is that you don't know the key when you're writing the JSLT? If that's the case you can use the `get-key` function:

    get-key(.product, $productId)
